### PR TITLE
feat: add rbs parser support

### PR DESCRIPTION
**Pull request by joker1007**
I would like to add rbs (Ruby type signature) parser support.

I see that there was a pull request in the past to support rbs, but it wasn't merged.
Sorry, this pull request is for a different implementation by myself.

### CODE REVIEW

File: queries/rbs/highlights.scm [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+(self) @variable.builtin
+
+(type (symbol_literal) @symbol)
+
+(type (string_literal (escape_sequence) @string.escape))
+(type (string_literal) @string)
+
+(type (integer_literal) @number)
+
+(ERROR) @error
```
Comment (lucario387) [inline queries/rbs/highlights.scm:85]: (ERROR) should now be on users' side, we don't really want the node here. See #5572

File: queries/rbs/highlights.scm [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  "alias"
+] @keyword
+
+[
+ "def"
+] @keyword.function
+
+(include_member "include" @function.method)  
+(extend_member "extend" @function.method)  
+(prepend_member "prepend" @function.method)
```
Comment (lucario387) [inline queries/rbs/highlights.scm:40]: Can't this be

```suggestion
[ "include" "extend" "prepend" ] @function.method
```

File: queries/rbs/highlights.scm [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+  (method_name
+    [
+     (identifier)
+     (constant)
+     (operator)
+     (setter)
+    ] @function.method))
+
+(ivar_member (ivar_name) @property)
+(ivar_member (cvar_name) @property)
```
Comment (lucario387) [inline queries/rbs/highlights.scm:56]: Is this also possible?

```suggestion
[ (ivar_name) (cvar_name) ] @property
```

**Comment by joker1007** [comment]
@lucario387 
Thank you very much for your reviews! It's so fast!
I have accepted your suggestions and have fixed them.

File: queries/rbs/highlights.scm [hunk 4]
```diff
@@ -0,0 +1,110 @@
+(use_directive
+ "use" @keyword)
+(use_clause
+ "as" @keyword)
```
Comment (lucario387) [inline queries/rbs/highlights.scm:4]: This can probably be reduced

**Comment by joker1007** [comment]
I addressed the next review comments. I have fixed them.
Thanks!!

File: queries/rbs/highlights.scm [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+(namespace (constant) @namespace)
+
+(builtin_type) @type.builtin
+
+(const_name (constant) @constant)
+(global_name) @property
+
+(parameter (var_name) @variable.parameter)
+
+(keyword) @variable.parameter
```
[the remark this answers is not in the document]
Comment (joker1007) [reply queries/rbs/highlights.scm:72]: It's not a keyword.
 `(keyword)` is a part of "keyword arguments" which is a feature in Ruby.
Probably, it's a `@parameter` token.

File: queries/rbs/highlights.scm [hunk 6]
```diff
@@ -67,9 +63,9 @@
 (const_name (constant) @constant)
 (global_name) @property
 
-(parameter (var_name) @variable.parameter)
+(parameter (var_name) @parameter)
 
-(keyword) @variable.parameter
+(keyword) @parameter
```
Comment (lucario387) [inline queries/rbs/highlights.scm:68]: I think having a comment for this case is kindof needed, just to clear away the misunderstandings

Something like `keyword arguments` as a comment right on top of this capture will be nice

Comment (joker1007) [reply queries/rbs/highlights.scm:68]: You are right. I have added comments.

**Comment by joker1007** [comment]
Thank you for the approval!

By the way, a test (terraform_spec) failed on CI. It does not appear to be related to this pull request.
I'm sorry, I don't know the cause.
Is it sometimes flaky?

**Pull request merged**